frontend/src/components/ProjectCardGrid/index.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect } from 'react'
+import { useInView } from 'react-intersection-observer'
+
+import { meiliIndex } from '@utils/meili'
+import { Filter } from 'meilisearch'
+import ProjectCard, { IntersectionObserverRef } from '../ProjectCard'
+import styles from './index.module.scss'
+
+export const cardsPerRow = 3
+const LIMIT = cardsPerRow * 6
+
+export interface Project {
+  bom: object
+  gitHash: string
+  id: string
+  ownerName: string
+  projectName: string
+  readme: string
+  repoId: string
+  repoName: string
+  summary: string
+}
+
+interface CardsGridProps {
+  projects: Project[]
+  intersectionObserverRef?: IntersectionObserverRef
+}
+
+const CardsGrid = ({ projects, intersectionObserverRef }: CardsGridProps) => {
+  return (
+    <div className={styles.cardsGrid} data-cy="cards-grid">
+      {projects?.map((project, index) => (
+        <ProjectCard
+          {...project}
+          key={project.id}
+          intersectionObserverRef={
+            index === projects.length - cardsPerRow * 2
+              ? intersectionObserverRef
+              : null
+          }
+        />
+      ))}
+    </div>
+  )
+}
+
+export const getKey =
+  ({ query, filter }: { query: string; filter?: Filter }) =>
+  (pageIndex: number, previousPageData: Project[]) => {
+    if (previousPageData && !previousPageData.length) {
+      // reached the end
+      return null
+    }
+    return { query, offset: pageIndex * LIMIT, limit: LIMIT, filter }
+  }
```

Comment:
The name `getKey` is confusing to me. Key of what? It seems to be returning an object.

Reply:
SWR keys can be objects, this is why we are using `unstable_serialize`.  And `getKey` is the name of this function in [SWRInfinite itself](https://swr.vercel.app/docs/pagination#parameters).

Reply:
Could you call it something like `getSwrKey` to match `swrFallback`?

Reply:
Resolved in 11b9354.

Reply:
I just noticed this is returning a function so I think it should be called something that makes that clear. `makeSWRInfiniteKeyGetter`?